src/github.com/couchbase/sync_gateway/rest/routing.go

```diff
@@ -251,6 +265,22 @@ func wrapRouter(sc *ServerContext, privs handlerPrivs, router *mux.Router) http.
 	})
 }
 
+func matchedOrigin(allowOrigins []string, rqOrigins []string) string {
+	for _, av := range allowOrigins {
+		for _, rv := range rqOrigins {
+			if rv == av {
+				return av
+			}
+		}
+	}
+	for _, av := range allowOrigins {
+		if av == "*" {
+			return "*"
+		}
+	}
+	return ""
+}
+
```

Comment:
Probably not yet perf critical for the reasons you raised. Do we have a mechanism to be notified of config changes? Can this config change without a restart?